core/base-service/base-toml.spec.js

```diff
@@ -0,0 +1,150 @@
+import Joi from 'joi'
+import { expect } from 'chai'
+import sinon from 'sinon'
+import BaseTomlService from './base-toml.js'
+
+const dummySchema = Joi.object({
+  requiredString: Joi.string().required(),
+}).required()
+
+class DummyTomlService extends BaseTomlService {
+  static category = 'cat'
+  static route = { base: 'foo' }
+
+  async handle() {
+    const { requiredString } = await this._requestToml({
+      schema: dummySchema,
+      url: 'http://example.com/foo.toml',
+    })
+    return { message: requiredString }
+  }
+}
+
+const expectedToml = `
+# example toml
+requiredString = "some-string"
+`
+
+const unexpectedToml = `
+# example toml - unexpected val
+unexpectedKey = "some-string"
+`
+
+const invalidToml = `
+# example invalid toml
+missing= "space"
+colonsCantBeUsed: 42
+missing "assignment"
+`
```

Comment:
I find the use of "expected" and "unexpected" a bit unclear here. Particularly given that in the context of unit tests, we'd also talk about the "expected" value as the thing we want to compare the observed value to.

I also appreciate that _naming things is hard_ here because we've got two very different "invalid" cases. There is:

- "invalid toml" - legal toml syntax that doesn't validate against the schema and
- "invalid toml" - illegal toml syntax that can't be parsed

I think we could probably express this more clearly by calling these 3

- `validToml`
- `invalidToml`
- `malformedToml`

or

- `validToml`
- `invalidToml`
- `unparseableToml`

Reply:
Solved with https://github.com/badges/shields/pull/9438/commits/f762a3d22600870bcd592bf2007456ec4219d038
I changed a bit the names, hope those make it as clear as it can get

`invalidSchemaToml` . `invalidTomlSyntax`

Reply:
Yes. Awesome - thanks